recipes/inkscape/wrapper.c

```diff
@@ -0,0 +1,262 @@
+    fprintf(stderr, format, data);
+    return 2;
+}
+
+
+char *quoted(char *data)
+{
+    int i, ln = strlen(data), nb;
+
+    /* We allocate twice as much space as needed to deal with worse-case
+       of having to escape everything. */
+    char *result = calloc(ln * 2 + 3, sizeof(char));
+    char *presult = result;
+
+    *presult++ = '"';
+    for (nb=0, i=0; i < ln; i++)
+    {
+        if (data[i] == '\\')
+            nb += 1;
+        else if (data[i] == '"')
+        {
+            for (; nb > 0; nb--)
+                *presult++ = '\\';
+            *presult++ = '\\';
+        }
+        else
+            nb = 0;
+        *presult++ = data[i];
+    }
+
+    for (; nb > 0; nb--)        /* Deal w trailing slashes */
+        *presult++ = '\\';
+
+    *presult++ = '"';
+    *presult++ = 0;
+    return result;
+}
+
+
+
+void pass_control_to_child(DWORD control_type)
+{
+    /*
+     * distribute-issue207
+     * passes the control event to child process (Python)
+     */
+    if (!child_pid) {
+        return;
+    }
+    GenerateConsoleCtrlEvent(child_pid, 0);
+}
+
+
+BOOL control_handler(DWORD control_type)
+{
+    /*
+     * distribute-issue207
+     * control event handler callback function
+     */
+    switch (control_type) {
+        case CTRL_C_EVENT:
+            pass_control_to_child(0);
+            break;
+    }
+    return TRUE;
+}
+
+
+int create_and_wait_for_subprocess(char* command) {
+    /*
+     * distribute-issue207
+     * launches child process (Python)
+     */
+    DWORD return_value = 0;
+    LPSTR commandline = command;
+    STARTUPINFOA s_info;
+    PROCESS_INFORMATION p_info;
+    ZeroMemory(&p_info, sizeof(p_info));
+    ZeroMemory(&s_info, sizeof(s_info));
+    s_info.cb = sizeof(STARTUPINFO);
+    // set-up control handler callback funciotn
+    SetConsoleCtrlHandler((PHANDLER_ROUTINE) control_handler, TRUE);
+    if (!CreateProcessA(NULL, commandline, NULL, NULL, TRUE, 0, NULL,
+                        NULL, &s_info, &p_info))
+    {
+        fprintf(stderr, "failed to create process.\n");
+        return 1;
+    }
+    child_pid = p_info.dwProcessId;
+    // wait for Python to exit
+    WaitForSingleObject(p_info.hProcess, INFINITE);
+    if (!GetExitCodeProcess(p_info.hProcess, &return_value))
+    {
+        fprintf(stderr, "failed to get exit code from process.\n");
+        return 0;
+    }
+    return return_value;
+}
+
+
+char* join_executable_and_args(char *executable, char **args, int argc)
+{
+    /*
+     * distribute-issue207
+     * CreateProcess needs a long string of the executable and command-line arguments,
+     * so we need to convert it from the args that was built
+     */
+    int len, counter;
+    char* cmdline;
+
+    len = strlen(executable) + 2;
+    for (counter=1; counter<argc; counter++)
+    {
+        len += strlen(args[counter]) + 1;
+    }
+
+    cmdline = (char *) calloc(len, sizeof(char));
+    sprintf(cmdline, "%s", executable);
```

Comment:
I went the other way and removed all strcpy/strcat and the filesize went down to 60kb instead of 120kb.